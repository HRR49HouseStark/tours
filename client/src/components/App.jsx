import React, { useState, useEffect } from 'react';

import { getData, submitForm } from '../utils/Logic';

import TourRequest from './TourRequest';
import TourType from './TourType';
import Calendar from './Calendar/Calendar';
import TimeDropdown from './Calendar/TimeDropdown';
import RequestForm from './Form/RequestForm';
import Disclaimer from './Disclaimer';
import AgentList from './Agents/AgentList';

const App = () => {
  // State for API calls
  const [requests, setRequests] = useState([]);
  const [agents, setAgents] = useState([]);

  // State for user inputs
  const [tour, toggleTour] = useState(true);
  const [digital, setDigital] = useState(false);
  const [currentDate, setDate] = useState('');
  const [call, setCall] = useState(false);
  const [time, setTime] = useState('');
  const [agent, setAgent] = useState('');

  useEffect(() => {
    getData(setRequests, setAgents);
  }, []);

  const submit = (form) => {
    const toSubmit = !tour ? { agent, ...form }
      : { date: currentDate, type: digital, time, ...form };
    submitForm({ call, ...toSubmit }).then(() => getData(setRequests, setAgents));
  };

  return (
    <div className="appContainer">
      <TourRequest tour={tour} toggleTour={toggleTour} />
      <div id="moduleContainer">
        {tour ? (
          <div>
            <TourType digital={digital} setDigital={setDigital} />
            <Calendar currentDate={currentDate} setDate={setDate} />
            <TimeDropdown occupied={requests} currentDate={currentDate} setTime={setTime} />
          </div>
        ) : null}
        <RequestForm tour={tour} call={call} setCall={setCall} submit={submit} />
        <Disclaimer tour={tour} />
        {!tour ? (<AgentList agents={agents} setAgent={setAgent} />) : null}
      </div>

    </div>
  );
};

export default App;
