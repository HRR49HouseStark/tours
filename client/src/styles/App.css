#app {
  border: solid 16px transparent;
  border-style: solid;
  border-color: transparent;
  border-width: 16px 8px 0px;
  box-sizing: border-box;
  display: block;

  /* CHANGE THESE FOR APP POSITIONING ON PAGE */
  position: absolute;
  width: 325px;
  right: 20%;
}

#module {
  display: flex;
  flex-direction: column;
  box-shadow: rgba(59, 65, 68, 0.18) 0px 17px 21px -1px;
  background-color: rgb(255, 255, 255);
  border: 1px solid rgb(232, 233, 234);
  padding: 16px;
  border-radius: 0px 8px 8px 8px;
  border-top: none;
}