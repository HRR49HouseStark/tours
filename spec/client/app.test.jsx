import React from 'react';
import { shallow } from 'enzyme';

import App from '../../client/src/components/App';

describe('App Test Suite', () => {
  const wrapper = shallow(<App />);

  test('Should render', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('should have a div with appContainer class', () => {
    const divs = wrapper.find('div');

    let found = false;

    divs.forEach((div) => {
      div.prop('className') === 'appContainer' ? found = true : null;
    });

    expect(found).toEqual(true);
  });

  test('snapshot render should match', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
