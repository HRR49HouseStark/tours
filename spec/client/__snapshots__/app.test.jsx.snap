// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Suite snapshot render should match 1`] = `
<div
  className="appContainer"
>
  <TourRequest
    toggleTour={[Function]}
    tour={true}
  />
  <div
    id="moduleContainer"
  >
    <div>
      <TourType
        digital={false}
        setDigital={[Function]}
      />
      <Calendar
        currentDate=""
        setDate={[Function]}
      />
      <TimeDropdown
        currentDate=""
        occupied={Array []}
        setTime={[Function]}
      />
    </div>
    <RequestForm
      call={false}
      setCall={[Function]}
      submit={[Function]}
      tour={true}
    />
    <Disclaimer
      tour={true}
    />
  </div>
</div>
`;
